src/solvers/vectorbundle_newton.jl

```diff
@@ -0,0 +1,340 @@
+
+# Example
+
+On an $(_link(:AbstractPowerManifold)) like ``$(_math(:M)) = $(_math(:M; M = "N"))^n``
+any point ``p = (p_1,…,p_n) ∈ $(_math(:M))`` is a vector of length ``n`` with of points ``p_i ∈ $(_math(:M; M = "N"))``.
+Then, denoting the `outer_norm` by ``r``, the distance of two points ``p,q ∈ $(_math(:M))``
+is given by
+
+```math
+$(_math(:distance))(p,q) = $(_tex(:Bigl))( $(_tex(:sum))_{k=1}^n $(_math(:distance))(p_k,q_k)^r $(_tex(:Bigr)))^{$(_tex(:frac, "1", "r"))},
+```
+
+where the sum turns into a maximum for the case ``r=∞``.
+The `outer_norm` has no effect on manifolds that do not consist of components.
+
+If the manifold does not have components, the outer norm is ignored.
+
+# Constructor
+
+    AffineCovariantStepsize(
+        M::AbstractManifold=DefaultManifold(2);
+        α=1.0, θ=1.3, θ_des=0.5, θ_acc=1.1*θ_des, outer_norm::Real=missing
+    )
+
+Initializes all fields, where none of them is mandatory. The length is set to ``1.0``.
+
+Since the computation of the convergence monitor ``θ`` requires simplified Newton directions a method for computing them has to be provided.
+This should be implemented as a method of the `newton_equation(M, VB, p, p_trial)` as parameters and returning a representation of the (transported) ``F(p_{$(_tex(:rm, "trial"))})``.
+"""
+mutable struct AffineCovariantStepsize{T, R <: Real, N <: Union{Real, Missing}} <: Stepsize
+    α::T
+    θ::R
+    θ_des::R
+    θ_acc::R
+    last_stepsize::R
+    outer_norm::N
+end
+function AffineCovariantStepsize(
+        M::AbstractManifold = DefaultManifold(2);
+        α = 1.0, θ = 1.3, θ_des = 0.5, θ_acc = 1.1 * θ_des, outer_norm::N = missing
+    ) where {N <: Union{Real, Missing}}
+    return AffineCovariantStepsize{typeof(α), typeof(θ), N}(α, θ, θ_des, θ_acc, 1.0, outer_norm)
+end
+
+function (acs::AffineCovariantStepsize)(
+        amp::AbstractManoptProblem, ams::VectorBundleNewtonState, ::Any, args...; kwargs...
+    )
+    α_new = copy(acs.α)
+    θ_new = copy(acs.θ)
+    b = copy(amp.newton_equation.b)
+    while θ_new > acs.θ_acc && α_new > 1.0e-10
+        acs.last_stepsize = copy(α_new)
+        Xα = α_new * ams.X
+        M = get_manifold(amp)
+        retract!(M, ams.p_trial, ams.p, Xα, ams.retraction_method)
+
+        rhs_next = amp.newton_equation(M, get_vectorbundle(amp), ams.p, ams.p_trial)
+        rhs_simplified = rhs_next - (1.0 - α_new) * b
+        amp.newton_equation.b .= rhs_simplified
+
+        simplified_newton = ams.sub_problem(amp, ams)
+
+        if has_components(M) && !ismissing(acs.outer_norm)
+            θ_new = norm(amp.manifold, ams.p, simplified_newton, acs.outer_norm) / norm(amp.manifold, ams.p, ams.X, acs.outer_norm)
+        else
+            θ_new = norm(simplified_newton) / norm(ams.X)
+        end
+
+        α_new = min(1.0, ((α_new * acs.θ_des) / θ_new))
+        if α_new < 1.0e-15
+            println("Newton's method failed")
+            return
+        end
```

Comment:
This is currently never reached. What happens if it does reach this? Currently it does nothing and would continue in the next iterate - to end up here again? Is that if necessary?